### PR TITLE
Merge multi-device tensor Llama into experimental folder

### PR DESCRIPTION
**Pull request by cglagovichTT**
Hard dependency on https://github.com/tenstorrent/tt-metal/pull/8521 to enable async create_heads_decode

### CODE REVIEW

**Comment by cglagovichTT** [comment]
fyi @xuncaiTT @caixunshiren @kevinmiTT11

**Comment by cglagovichTT** [comment]
We want to get this in asap so we can start running perf CI on it. Shouldn't break any existing tests since experimental isn't yet in CI

File: conftest.py [hunk 1]
```diff
@@ -389,6 +389,13 @@ def t3k_device_mesh(request, silicon_arch_name, silicon_arch_wormhole_b0):
     logger.debug(f"multidevice with {device_mesh.get_num_devices()} devices is created")
     yield device_mesh
 
+    import tt_lib as ttl
+
+    for i in device_mesh.get_device_ids():
+        device = device_mesh.get_device(i)
+        ttl.device.DumpDeviceProfiler(device, True)
```
Comment (tt-rkim) [inline conftest.py:396]: This is going to run for every test, hope this is ok
cc: @tapspatel @mo-tenstorrent

**Review by tt-rkim — APPROVED** [review]

**Comment by tt-rkim** [comment]
I don't see a perf test here

Are you actively working on a perf test for llama, or not started yet?

**Comment by cglagovichTT** [comment]
> I don't see a perf test here
> 
> Are you actively working on a perf test for llama, or not started yet?

Actively working on. I want to merge this PR after
https://github.com/tenstorrent/tt-metal/pull/8521 and https://github.com/tenstorrent/tt-metal/pull/8480 merge

File: conftest.py (hunk 1 above)
Comment (mo-tenstorrent) [reply conftest.py:396]: Yeah, in normal builds this is a no-op

**Pull request merged**